.github/workflows/test.yml

```diff
@@ -2,9 +2,7 @@ name: test
 
 on:
   push:
-    branches: [ "main" ]
```

Comment:
You probably want to keep it here - otherwise you effectively double builds:
* it builds when you push to a fork branch (e.g. most forks will automatically build it when you initially push your code)
* it also builds on pull request

Instead, I would recommend keeping the list of `push` branches to the officially maintained ones (e.g. `main`, `v1.x`, `complex-feature`, etc), but pull requests could in theory be built for everything (i.e. it will build if I create a PR into one of your branch)...     But, tbh, i have not seen a case when these settings didn't work for a project, but there is always a new case of course.

Reply:
Thanks for your thoughts. What I'm after is:

 - builds on pushes to any branch, even forks (now that we are attempting to work in forks)
 - Builds of the merge commit for a PR to branches that are non main ([feature/update-js-parser](https://github.com/maplibre/maplibre-tile-spec/tree/feature/update-js-parser) but also potentially PRs stacked on top of PRs targeting `[feature/update-js-parser]`

Reply:
Restricted back to main in 7c11123. I think I'll try using manual builds with `workflow_dispatch` for any situations that come up where a CI run is useful and a PR does not exist.

Reply:
Just to understand - you want github action to run on every one of these steps?
* you push your code to your own fork, e.g. `springmeyer:new-feature`
* you create a pull request, attempting to merge `springmeyer:new-feature` into `maplibre:main` or into `maplibre:feature/update-js-parser`
* you merge changes into either of the two branches

The first two are usually identical - so you just run the same CI twice -- once in the context of your own fork, and once in the context of the primary one (but without access to the primary's secrets). That's why in most repos CI runs only on the specific primary branches like `main` and `feature/update-js-parser`, but nothing else.  PRs on the other hand can run on everything.

Note that pushing your changes to your own fork is no different than pushing it to the primary repo -- CI would still not have ran with the current settings. It only ran after you created a PR.

Reply:
@springmeyer just out of curiosity - do you have an example of such case?  I personally never seen a branch on a fork that has no PR into one of the primary repo's branches, so really curious. Thx!